server/src/main/java/org/elasticsearch/action/bulk/Retry.java

```diff
@@ -48,18 +48,32 @@ public Retry(BackoffPolicy backoffPolicy, Scheduler scheduler) {
         this.scheduler = scheduler;
     }
 
+    /**
+     * Invokes #accept(BulkRequest, ActionListener). Backs off on the provided exception and delegates results to the
+     * provided listener. Retries will be scheduled using the class's thread pool.
+     * @param consumer The consumer to which apply the request and listener
+     * @param bulkRequest The bulk request that should be executed.
+     * @param listener A listener that is invoked when the bulk request finishes or completes with an exception. The listener is not
+     */
+    public void withBackoff(BiConsumer<BulkRequest, ActionListener<BulkResponse>> consumer, BulkRequest bulkRequest,
```

Comment:
Since this is actively part of the public API I'm deprecating the old way before removing it in master.